packages/core/contracts/optimistic-asserter/implementation/escalation-manager/DisputeLimitingEscalationManager.sol

```diff
@@ -0,0 +1,71 @@
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./BaseEscalationManager.sol";
+import "../../interfaces/OptimisticAsserterInterface.sol";
+
+// This Escalation Manager blocks all assertions till the blocking dispute is resolved by Oracle. In order to avoid
+// interference among different applications this Escalation Manager allows assertions only from one requesting contract.
+// This is useful to create a system where only one assertion dispute can occur at a time.
+contract DisputeLimitingEscalationManager is BaseEscalationManager, Ownable {
+    OptimisticAsserterInterface public immutable optimisticAsserter;
+
+    // Address of linked requesting contract. Before this is set via setAssertingCaller all assertions will be blocked.
+    address public assertingCaller;
+
+    bytes32 public disputedAssertionId;
+
+    event AssertingCallerSet(address indexed assertingCaller);
+
+    constructor(address _optimisticAsserter) {
+        optimisticAsserter = OptimisticAsserterInterface(_optimisticAsserter);
+    }
+
+    // Set the address of the contract that will be allowed to use Optimistic Asserter.
+    // This can only be set once. We do not set this at constructor just to allow for some flexibility in the ordering
+    // of how contracts are deployed.
+    function setAssertingCaller(address _assertingCaller) public onlyOwner {
+        require(_assertingCaller != address(0), "Invalid asserting caller");
+        require(assertingCaller == address(0), "Asserting caller already set");
+        assertingCaller = _assertingCaller;
+        emit AssertingCallerSet(_assertingCaller);
+    }
+
+    function getAssertionPolicy(bytes32 assertionId) public view override returns (AssertionPolicy memory) {
+        OptimisticAsserterInterface.Assertion memory assertion = optimisticAsserter.getAssertion(assertionId);
+        return
+            AssertionPolicy({
+                blockAssertion: _checkIfShouldBlockAssertion(assertion),
```

Comment:
another nit :)

there is no point in passing `assertion`. you might as well just pass `assertionId` and then `_checkIfShouldBlockAssertion` just takes in a `bytes` and reads the state of the `assertingCaller` within that function.

Reply:
thanks, totally agree and changed to pass `assertionId`